dcos/mesos.py

```diff
@@ -335,19 +483,224 @@ def user(self):
 
         return self.framework()['user']
 
+    def executor(self):
+        """ Returns this tasks' executor
+
+        :returns: task's executor
+        :rtype: dict
+        """
+        for executor in self.slave().executor_dicts():
+            tasks = _merge(executor,
+                           ['completed_tasks',
+                            'tasks',
+                            'queued_tasks'])
+            if any(task['id'] == self['id'] for task in tasks):
+                return executor
+        return None
+
+    def directory(self):
+        """ Sandbox directory for this task
+
+        :returns: path to task's sandbox
+        :rtype: str
+        """
+        return self.executor()['directory']
```

Comment:
executors should have a 'directory' entry.  if they don't, that's an invalid state and we want to fail fast